### DISCUSSION TITLE
BKool app is reporting heart rate with a 1280 offset

### DISCUSSION THREAD

**Discussion by benriou** [cagnulein/qdomyos-zwift#207] [Q&A] [answered]
The BKOOL cycling app reports the heart rate from FTMS signal but with a 1280 positive offset.
The problem might be from the application to require a 16 bits value instead of 8 bit value for this variable. 

Is there any application reading correctly the heart rate sent by QZ via the FTMS data stream ?

**Comment by cagnulein** [comment]
nrfconnect app for sure :)
but please double check it ;)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 mar 2021 alle ore 12:15 ben75020 ***@***.***>
ha scritto:

**Comment by benriou** [comment, marked as answer]
BKOOL is now reporting the correct value after the following change. 

I searched with the NRF toolbox suite, but there is no bike detected when the Domyos bridge is started.

I'm ok to submit a pull request, however I don't want to break something on others users.

```
+++ b/src/virtualbike.cpp
@@ -496,7 +496,9 @@ void virtualbike::bikeProvider()
             value.append((char)(Bike->watts() & 0xFF)); // watts
             value.append((char)(Bike->watts() >> 8) & 0xFF); // watts
 
-            value.append(char(Bike->currentHeart().value())); // Actual value.
+            value.append(char(Bike->currentHeart().value()));
+           value.append((char)0);
```

**Comment by cagnulein** [comment]
Great make a PR, i will test it before pushing it.
Thanks

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 23 mar 2021 alle ore 13:39 ben75020 ***@***.***>
ha scritto:

**Comment by benriou** [comment]
done. this discussion can be closed (following on the PR)

**Comment by cagnulein** [comment]
Thank you for your great job!
